repository/src/main/java/com/epam/esm/repository/repositoryinterfaces/BaseRepository.java

```diff
@@ -0,0 +1,37 @@
+package com.epam.esm.repository.repositoryinterfaces;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.Tag;
+
+import java.util.Optional;
+
+/**
+ * The base interface for all repositories.
+ *
+ * @param <T> type of Entity
+ */
+public interface BaseRepository<T> {
```

Comment:
We don't use _Base_ prefix for interfaces and abstract classes. _Base_ means that class can be used and it is default implementation of interface.

Reply:
You can simply call it _Repository_

Reply:
fixed